xdsl/irdl/declarative_assembly_format.py

```diff
@@ -738,29 +775,6 @@ class OptionalAttributeVariable(AttributeVariable, OptionalVariable):
     The directive will request a space to be printed after.
     """
 
-    def parse_optional(self, parser: Parser, state: ParsingState) -> bool:
-        if attribute := parser.parse_optional_attribute():
-            if self.is_property:
-                state.properties[self.attr_name] = attribute
-            else:
-                state.attributes[self.attr_name] = attribute
-        return attribute is not None
-
-    def print(self, printer: Printer, state: PrintingState, op: IRDLOperation) -> None:
-        if state.should_emit_space or not state.last_was_punctuation:
-            printer.print(" ")
-        if self.is_property:
-            attr = op.properties.get(self.attr_name)
-        else:
-            attr = op.attributes.get(self.attr_name)
-        if attr:
-            printer.print_attribute(attr)
-            state.should_emit_space = True
-            state.last_was_punctuation = False
-
-    def is_present(self, op: IRDLOperation) -> bool:
-        return getattr(op, self.attr_name) is not None
-
```

Comment:
Not sure I understand how this is related, just incidentally removing duplicate code?

Reply:
Exactly!